move topk to use OpInfo

albanD opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#55547 move topk to use OpInfo**
* #55535 move logaddexp{2} to opinfo
* #55518 port addcdiv to opinfo
* #55517 port addcmul to OpInfo

Differential Revision: [D27649412](https://our.internmc.facebook.com/intern/diff/D27649412)

albanD committed:
move topk to use OpInfo

[ghstack-poisoned]

mruberry commented on torch/testing/_internal/common_methods_invocations.py:
```diff
@@ -3318,6 +3342,14 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
+           sample_inputs_func=sample_inputs_topk,
+           skips=(
+               # Topk is not raising a warning when the out is resized
+               SkipInfo('TestCommon', 'test_out'),
```
Nice skip + comment

mruberry commented on torch/testing/_internal/common_methods_invocations.py:
```diff
@@ -1078,6 +1078,30 @@ def sample_inputs_max_min_reduction_no_dim(op_info, device, dtype, requires_grad
+    inputs = []
+    inputs.append(SampleInput(get_tensor_input((S, M, S)), args=(3,)))
+    inputs.append(SampleInput(get_tensor_input((S, M, S)), args=(3, 1)))
+    inputs.append(SampleInput(get_tensor_input((S, M, S)), args=(3, -2)))
```
Nice new case

albanD replied:
These are not new.
The old spec was specifying which args were dimension arguments and were automatically testing `val` and `val - size()` to make sure we do the wrapping properly.
I didn't find an arg in OpInfo to specify that so I added the tests manually.

mruberry replied:
Cool!

mruberry reviewed (approved):
This is great! And thank you for adding some new cases, too!

albanD committed:
Update on "move topk to use OpInfo"




[ghstack-poisoned]

albanD committed:
Update on "move topk to use OpInfo"


Differential Revision: [D27649412](https://our.internmc.facebook.com/intern/diff/D27649412)

[ghstack-poisoned]

albanD committed:
Update on "move topk to use OpInfo"


Differential Revision: [D27649412](https://our.internmc.facebook.com/intern/diff/D27649412)

[ghstack-poisoned]

albanD committed:
Update on "move topk to use OpInfo"


Differential Revision: [D27649412](https://our.internmc.facebook.com/intern/diff/D27649412)

[ghstack-poisoned]
